stdlib/LibGit2/src/callbacks.jl

```diff
@@ -91,12 +91,15 @@ function authenticate_ssh(libgit2credptr::Ptr{Ptr{Cvoid}}, p::CredentialPayload,
             cred.user = unsafe_string(username_ptr)
         end
 
-        cred.prvkey = Base.get(ENV, "SSH_KEY_PATH") do
-            default = joinpath(homedir(), ".ssh", "id_rsa")
-            if isempty(cred.prvkey) && isfile(default)
-                default
-            else
-                cred.prvkey
+        if haskey(ENV, "SSH_KEY_PATH")
+            cred.prvkey = ENV["SSH_KEY_PATH"]
+        elseif isempty(cred.prvkey)
+            for keytype in ("rsa", "ecdsa")
```

Comment:
I thought about it but it may be considered breaking for people who are using RSA keys with Julia (sufficiently old ones still work with GitHub and other services may accept them) and also have an ECDSA key for other purposes but not registered where Julia is used (or just have one but doesn't use it).

I have no strong opinion whether that outweighs the advantage for many people of having an ECDSA key chosen over an RSA key. Either way I guess it needs to be communicated.

Reply:
FWIW ssh will try pretty much anything. You can confirm this with `ssh -vvvv`. may be good to also have `ed25519`, `id_ecdsa_sk`, and `id_ed25519_sk` on here as well.

Reply:
Command line ssh is quite capable, yes, but this code is for LibGit2/LibSSH2. Do you have any indications that those ciphers will help here?

Reply:
No, I misread: 
> Arguably the list of keys should be longer but at this point RSA and ECDSA keys are the only ones Julia's LibSSH2 understands (up to 1.7, RSA only).

:)